### PR TITLE
Add support for timezone on datetime list format

### PR DESCRIPTION
**Pull request by lightglitch**

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Can you please explain a little bit more with use case and, at best, some code sample and screenshots?

Thanks.

**Comment by lightglitch** [comment]
@Soullivaneuh You want me to explain here or to added it to the documentation?

**Comment by Soullivaneuh** [comment]
@lightglitch Both of them if possible. :-)

**Comment by lightglitch** [comment]
This is to support the correct visualization of the object local date by the user. 
If you have a system that stores all dates in UTC, when a user is editing objects he want's to see his local date not the UTC date.
In forms we use http://symfony.com/doc/current/reference/forms/types/datetime.html#view-timezone but in the list there is nothing to support that.

**Comment by Soullivaneuh** [comment]
@lightglitch I would like to test it. Can you give a use case of your feature?

Thanks.

**Comment by lightglitch** [comment]
@Soullivaneuh Just added a little more documentation, but use case is very simple like I explain in the previous comment and in the docs. The goal is to have two end users in different timezones to see the date in their local time, instead in UTC (if for example is stored in UTC).

**Comment by greg0ire** [comment]
Could you fix your commit messages ? They are a bit vague. Here is how they could look : 

The doc commits : 

> document something

The final commit : 

> Add support for timezone on datetime list format
>
> If you have a system that stores all dates in UTC, when a user is editing objects, she wants to see his local date, not the UTC date.

In general, never do commit messages that read "update path/to/file.extension". You are just paraphrasing git.

File: Resources/doc/reference/field_types.rst [hunk 1]
```diff
@@ -34,6 +34,17 @@ This is currently limited to scalar types (text, integer, url...).
 
     In ``date`` and ``datetime`` field types, ``format`` pattern must match twig's
     ``date`` filter specification, available at: `http://twig.sensiolabs.org/doc/filters/date.html <http://twig.sensiolabs.org/doc/filters/date.html>`_
+
+    In ``datetime`` field types, ``timezone`` syntax must match twig's
+    ``date`` filter specification, available at: `http://twig.sensiolabs.org/doc/filters/date.html <http://twig.sensiolabs.org/doc/filters/date.html>`_
+    and php timezone list: `https://php.net/manual/en/timezones.php <https://php.net/manual/en/timezones.php>`_
+    The goal is to have in lists what `view-timezone <http://symfony.com/doc/current/reference/forms/types/datetime.html#view-timezone>`_ allows on forms,
```
Comment (greg0ire) [inline Resources/doc/reference/field_types.rst]: The goal of what ? Your PR ? This is documentation, so you should write it as if your PR was already merged.

> You can use in lists what…

**Comment by lightglitch** [comment]
I'm a little busy this week but I will fix it as soon as possible.

**Comment by lightglitch** [comment]
@greg0ire done

**Comment by greg0ire** [comment]
Way better, thanks a lot ?

**Comment by lightglitch** [comment]
@Soullivaneuh I think is good to go.

**Comment by Soullivaneuh** [comment]
@lightglitch Could you rebase your PR from master?

I get a strange error when trying to install your fork:

```
PHP Fatal error:  Interface 'Sonata\AdminBundle\Model\LockInterface' not found
```

**Comment by lightglitch** [comment]
@Soullivaneuh The rebase added merges and other commits to this pull request I created a new one #3257 and will close this one.

**Comment by Soullivaneuh** [comment]
Just a `git rebase master` is sufficient, not merging. ;-)

**Comment by lightglitch** [comment]
@Soullivaneuh  That was what I did I don't know what happen.

**Pull request closed without merging**